ports/servoshell/prefs.rs

```diff
@@ -181,393 +182,228 @@ pub fn read_prefs_map(txt: &str) -> HashMap<String, PrefValue> {
+    layout_threads: Option<i64>,
 
-    // Make sure the default window size is not larger than any provided screen size.
-    let default_window_size = Size2D::new(1024, 740);
-    let default_window_size = screen_size_override
-        .map_or(default_window_size, |screen_size_override| {
-            default_window_size.min(screen_size_override)
-        });
-    let initial_window_size = opt_match
-        .opt_str("window-size")
-        .map_or(default_window_size, parse_resolution_string);
+    /// Directory root with unminified scripts
+    local_script_source: Option<String>,
 
-    let user_stylesheets = opt_match
-        .opt_strs("user-stylesheet")
-        .iter()
-        .map(|filename| {
-            let cwd = env::current_dir().unwrap();
-            let path = cwd.join(filename);
-            let url = ServoUrl::from_url(Url::from_file_path(&path).unwrap());
-            let mut contents = Vec::new();
-            File::open(path)
-                .unwrap_or_else(|err| args_fail(&format!("Couldn't open {}: {}", filename, err)))
-                .read_to_end(&mut contents)
-                .unwrap_or_else(|err| args_fail(&format!("Couldn't read {}: {}", filename, err)));
-            (contents, url)
-        })
-        .collect();
+    #[cfg(target_env = "ohos")]
+    /// Define a custom filter for logging
+    #[bpaf(argument("FILTER"))]
+    log_filter: Option<String>,
+
+    /// Run in multiprocess mode
+    #[bpaf(short('M'), long, fallback(true))]
+    multiprocess: bool,
+
+    /// Enable to turn off incremental layout
+    #[bpaf(short('i'), long, flag(false, true))]
+    nonincremental_layout: bool,
+
+    /// Path to an output image. The format of the image is determined
+    /// by the extension. Supports all formats that rust-image does.
+    #[bpaf(short('o'), long)]
+    output_image: Option<String>,
+
+    /// Time profiler flag and either a TSV output filename OR and interval
+    /// for output to Stdout (blank for Stdout with interval of 5s)
+    #[bpaf(short('p'), argument::<String>("10 OR time.tsv"), parse(parse_profiling), fallback(None))]
+    profile: Option<OutputOptions>,
+
+    /// Path to dump a self-contained HTML timeline of profiler traces
+    #[bpaf(long("profiler-trace-path"))]
+    time_profiler_trace_path: Option<String>,
+
+    /// A preference to set
+    #[bpaf(argument("dom_bluetooth_enabled"),fallback(vec![]))]
+    pref: Vec<String>,
+
+    /// Load an additional prefs from a file.
+    #[bpaf(long, argument("/path/to/prefs.json"), fallback(vec![]))]
+    prefs_files: Vec<String>,
+
+    print_pwm: bool,
+
+    /// Probability of randomly closing a pipeline (for testing constellation hardening)
+    #[bpaf(argument("0.25"))]
+    random_pipeline_closure_probability: Option<f32>,
+
+    /// A fixed seed for repeatability of random pipeline closure.
+    random_pipeline_closure_seed: Option<usize>,
+
+    /// Run in sandbox if multiprocess
+    #[bpaf(short('S'), long, fallback(true))]
```

Comment:
`fallback(true)` won't do anything here. For inverted `switch` you can use `flag` [1] with arguments flipped.

1. https://docs.rs/bpaf/latest/bpaf/params/struct.NamedArg.html#method.flag